File: src/app/classes/order.ts
import {ShippingDetails} from './shipping-details';
import {Product} from './product';

export class Order {

  _orderId: number;
  _customerUserName: string;
  _orderDate: Date;
  _orderStatus: string;
  _totalAmount: number;
  _address: ShippingDetails;
  _product: Product[];

  get product(): Product[] {
    return this._product;
  }

  set product(value: Product[]) {
    this._product = value;
  }

  get address(): ShippingDetails {
    return this._address;
  }

  set address(value: ShippingDetails) {
    this._address = value;
  }

// private List<Product> product = new ArrayList<Product>();
  get orderId(): number {
    return this._orderId;
  }

  set orderId(value: number) {
    this._orderId = value;
  }

  get customerUserName(): string {
    return this._customerUserName;
  }

  set customerUserName(value: string) {
    this._customerUserName = value;
  }

  get orderDate(): Date {
    return this._orderDate;
  }

  set orderDate(value: Date) {
    this._orderDate = value;
  }

  get orderStatus(): string {
    return this._orderStatus;
  }

  set orderStatus(value: string) {
    this._orderStatus = value;
  }

  get totalAmount(): number {
    return this._totalAmount;
  }

  set totalAmount(value: number) {
    this._totalAmount = value;
  }

}

File: backend/Ezeeshop-CustomerService/Ezeeshop-OrderServiceProducer/src/main/java/com/ezeeshop/entity/Order.java
package com.ezeeshop.entity;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
@ApiModel(description="Model Class of Order Table")
public class Order {
	@ApiModelProperty(notes = "Id of Order")
	private Long orderId;
	
	@ApiModelProperty(notes = "CustomerUserName to which Order is associated")
	private String customerUserName;
	
	@ApiModelProperty(notes = "OrderDate of Order")
	private LocalDate orderDate;
	
	@ApiModelProperty(notes = "OrderStatus of Order")
	private String orderStatus;
	
	@ApiModelProperty(notes = "Total amount of Order")
	private int totalAmount;

	@ApiModelProperty(notes = "Shipping Details to which order is shipped")
	private ShippingDetails address;
	
	@ApiModelProperty(notes = "List of products in particular Order")
	private List<Product> product = new ArrayList<Product>();

	
	public int getTotalAmount() {
		return totalAmount;
	}

	public void setTotalAmount(int totalAmount) {
		this.totalAmount = totalAmount;
	}

	public Long getOrderId() {
		return orderId;
	}

	public void setOrderId(Long orderId) {
		this.orderId = orderId;
	}

	public String getCustomerUserName() {
		return customerUserName;
	}

	public void setCustomerUserName(String customerUserName) {
		this.customerUserName = customerUserName;
	}

	public LocalDate getOrderDate() {
		return orderDate;
	}

	public void setOrderDate(LocalDate orderDate) {
		this.orderDate = orderDate;
	}
	

	public String getOrderStatus() {
		return orderStatus;
	}

	public void setOrderStatus(String orderStatus) {
		this.orderStatus = orderStatus;
	}

	public List<Product> getProduct() {
		return product;
	}

	public void setProduct(List<Product> product) {
		this.product = product;
	}
	public void addProduct(Product product) {
		this.product.add(product);
	}

	public ShippingDetails getAddress() {
		return address;
	}

	public void setAddress(ShippingDetails address) {
		this.address = address;
	}
	
	
	
}
